File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.java
class Solution {
    public String firstPalindrome(String[] words) {
        String str = "";
        for(int i = 0; i < words.length; i++){
            str = words[i];
            if(str.length() > 0){
                int l = 0;
                int r = str.length()-1;
                if(str.length() == 1) return str;
                while(l < r){
                    if(str.charAt(l) == str.charAt(r)){
                        l++;
                        r--;
                    } else{
                        break;
                    }
                    if(l == r) return str;
                }
            }
        }
        return "";
    }
}

File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.cpp
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
        string str = "";
        for (int i = 0; i < words.size(); i++) {
            str = words[i];
            if(str.size() > 0) {
                int left = 0;
                int right = str.size()-1;
                if(str.size() == 1) return str;
                while(left < right) {
                    if(str[left] == str[right]) {
                        left++;
                        right--;
                    } else {
                        break;
                    }
                }
                if(left == right || left > right) return str;
            }
        }
        return "";
    }
};